### PR TITLE
test(planning_test_utils): create route and odometry from lane_id

### PR DESCRIPTION
**Pull request by kyoichi-sugahara**
## Description

Related with https://github.com/autowarefoundation/autoware.universe/issues/3723

create odometry from lane_id and make route from start pose and goal pose which derived from lane_id.
In this PR, route for pull put is implemented.
And shoulder lane is added for the map.

## Tests performed

Not applicable.

## Effects on system behavior

Not applicable.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/planning_test_utils/include/planning_interface_test_manager/planning_interface_test_manager_utils.hpp [hunk 1]
```diff
@@ -290,6 +269,127 @@ Scenario makeScenarioMsg(const std::string scenario)
...
+Odometry makeInitialPoseFromLaneId(const int & lane_id)
+{
+  Odometry current_odometry;
+  current_odometry.pose.pose = createPoseFromLaneID(lane_id);
+  current_odometry.header.frame_id = "map";
+
+  return current_odometry;
+}
+
+RouteSections combine_consecutive_route_sections(
```
Comment (TakaHoribe) [inline planning/planning_test_utils/include/planning_interface_test_manager/planning_interface_test_manager_utils.hpp:312]: This should be Camel case

File: planning/planning_test_utils/include/planning_interface_test_manager/planning_interface_test_manager_utils.hpp [hunk 2]
```diff
@@ -290,6 +269,127 @@ Scenario makeScenarioMsg(const std::string scenario)
...
+    // remove end route section because it is overlapping with first one in next route_section
+    route_sections.insert(route_sections.end(), route_sections1.begin(), route_sections1.end() - 1);
+  }
+  if (!route_sections2.empty()) {
+    route_sections.insert(route_sections.end(), route_sections2.begin(), route_sections2.end());
+  }
+  return route_sections;
+}
+
+Route::Message makeBehaviorRouteFromLaneId(const int & start_lane_id, const int & goal_lane_id)
```
Comment (TakaHoribe) [inline planning/planning_test_utils/include/planning_interface_test_manager/planning_interface_test_manager_utils.hpp:327]: Please write a description for this function. To explain the strategy to make the route.

File: planning/planning_test_utils/include/planning_interface_test_manager/planning_interface_test_manager.hpp [hunk 3]
```diff
@@ -252,7 +254,8 @@ class PlanningInterfaceTestManager
   void publishAbnormalRoute(
     rclcpp::Node::SharedPtr target_node, const LaneletRoute & abnormal_route);
 
-  void publishBehaviorNominalRoute(rclcpp::Node::SharedPtr target_node, std::string topic_name);
+  void publishBehaviorNominalRoute(
+    rclcpp::Node::SharedPtr target_node, std::string topic_name, std::string module_name = "");
```
Comment (TakaHoribe) [inline planning/planning_test_utils/include/planning_interface_test_manager/planning_interface_test_manager.hpp:258]: Could it be better to use an enum?

File: planning/planning_test_utils/include/planning_interface_test_manager/planning_interface_test_manager_utils.hpp (hunk 2 above)
Comment (kyoichi-sugahara) [reply planning/planning_test_utils/include/planning_interface_test_manager/planning_interface_test_manager_utils.hpp:329]: @TakaHoribe 
I added the description I hope it's enough but if not, I'd be happy to add more:smile:

Comment (TakaHoribe) [reply planning/planning_test_utils/include/planning_interface_test_manager/planning_interface_test_manager_utils.hpp]: Great! Thanks!

**Review by TakaHoribe — APPROVED** [review]
LGTM

**Pull request merged**